fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraNodes.java

```diff
@@ -125,7 +124,9 @@ public Response copyObject(@HeaderParam("Destination") final String destinationU
             LOGGER.info("Copy from '{}' to '{}'", source, destination);
             nodeService.copyObject(session, source, destination);
 
-            session.save();
+            if (!batchService.exists(session.getId(), getUserPrincipal())) {
```

Comment:
As indicated above, with runtime polymorphism, this awkward construct will go away.